extensions/web/lib.rs

```diff
@@ -15,12 +34,197 @@ pub fn init() -> Extension {
       "02_event.js",
       "03_abort_signal.js",
       "04_global_interfaces.js",
+      "05_base64.js",
       "08_text_encoding.js",
       "12_location.js",
     ))
+    .ops(vec![
+      ("op_base64_decode", op_sync(op_base64_decode)),
+      ("op_base64_encode", op_sync(op_base64_encode)),
+      (
+        "op_encoding_normalize_label",
+        op_sync(op_encoding_normalize_label),
+      ),
+      ("op_encoding_new_decoder", op_sync(op_encoding_new_decoder)),
+      ("op_encoding_decode", op_sync(op_encoding_decode)),
+      ("op_encoding_encode_into", op_sync(op_encoding_encode_into)),
+    ])
     .build()
 }
 
+fn op_base64_decode(
+  _state: &mut OpState,
+  s: String,
+  _: (),
+) -> Result<ZeroCopyBuf, AnyError> {
+  let cfg = base64::Config::new(base64::CharacterSet::Standard, true)
+    .decode_allow_trailing_bits(true);
+  let out = base64::decode_config(&s, cfg)?;
+  Ok(ZeroCopyBuf::from(out))
+}
+
+fn op_base64_encode(
+  _state: &mut OpState,
+  s: ZeroCopyBuf,
+  _: (),
+) -> Result<String, AnyError> {
+  let cfg = base64::Config::new(base64::CharacterSet::Standard, true)
+    .decode_allow_trailing_bits(true);
+  let out = base64::encode_config(&s, cfg);
+  Ok(out)
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct DecoderOptions {
+  label: String,
+  ignore_bom: bool,
+  fatal: bool,
+}
+
+fn op_encoding_normalize_label(
+  _state: &mut OpState,
+  label: String,
+  _: (),
+) -> Result<String, AnyError> {
+  let encoding = Encoding::for_label(label.as_bytes()).ok_or_else(|| {
```

Comment:
You might want to use [`Encoding::for_label_no_replacement`](https://docs.rs/encoding_rs/0.8.28/encoding_rs/struct.Encoding.html#method.for_label_no_replacement) here rather than detecting the replacement encoding in JS.

Reply:
Done, thanks.